### PR TITLE
fix: add password confirmation to change admin settings

### PR DESCRIPTION
**Pull request by janepie**

### CODE REVIEW

File: lib/Controller/SettingsController.php [hunk 1]
```diff
@@ -73,6 +74,7 @@ public function isDiscoveryEndpointValid($url) {
 		return $result;
 	}
 
+	#[PasswordConfirmationRequired]
```
Comment (julien-nc) [inline lib/Controller/SettingsController.php:77]: `OCP\AppFramework\Http\Attribute\PasswordConfirmationRequired` is only available in NC >= 27. You can use the old annotation instead:

```suggestion
	/**
	 * @PasswordConfirmationRequired
	 */
```

**Review by julien-nc — CHANGES_REQUESTED** [review]

File: lib/Controller/SettingsController.php (hunk 1 above)
Comment (janepie) [reply lib/Controller/SettingsController.php]: > `OCP\AppFramework\Http\Attribute\PasswordConfirmationRequired` is only available in NC >= 27. You can use the old annotation instead:

Ah yes, got it, thank you! Not sure about the failing integration checks though, seems to be something with the php version?

File: lib/Controller/SettingsController.php [hunk 2]
```diff
@@ -17,6 +17,7 @@
 use OCP\AppFramework\Controller;
 use OCP\AppFramework\Db\DoesNotExistException;
 use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\PasswordConfirmationRequired;
```
Comment (julien-nc) [inline lib/Controller/SettingsController.php:20]: This line should be deleted :wink:.
After having committed this change, you can squash the 3 last commits together with `git rebase -i HEAD~3` and force push.

**Review by julien-nc — CHANGES_REQUESTED** [review]

File: lib/Controller/SettingsController.php (hunk 1 above)
Comment (julien-nc) [reply lib/Controller/SettingsController.php]: Yeah we can ignore those until we figure out how to fix them. The wrong version of Php is installed in there.

**Review by julien-nc — APPROVED** [review]
Nice interactive rebase!

**Review by edward-ly — APPROVED** [review]
Seems fine with me too, although some commit messages adhere to the [Conventional Commits spec](https://www.conventionalcommits.org/en/v1.0.0/) while others don't. Either now or in the future, choose whether you want to adhere to it or not and stick with it.

**Pull request merged**